web/src/lib/components/asset-viewer/detail-panel-rating.svelte

```diff
@@ -0,0 +1,102 @@
+<script lang="ts">
+  import { handleError } from '$lib/utils/handle-error';
+  import { updateAsset, type AssetResponseDto } from '@immich/sdk';
+  import { t } from 'svelte-i18n';
+
+  export let asset: AssetResponseDto;
+  export let isOwner: boolean;
+
+  const countStars = 5;
+
+  $: rating = asset.exifInfo?.rating || 0;
+  $: currentRating = rating;
+
+  const handleChangeRating = async (newRating: number) => {
+    try {
+      await updateAsset({ id: asset.id, updateAssetDto: { rating: newRating } });
+    } catch (error) {
+      handleError(error, $t('cant_apply_changes'));
+    }
+    rating = currentRating;
+  };
+
+  const onChangeRating = () => {
```

Comment:
Pretty sure you can also pass an async function to `on:change`, so you could `await handleChangeRating`.

Reply:
That's right, now onChangeRating is no longer needed